aws/eks/main.tf

```diff
@@ -1,46 +1,73 @@
-data "aws_subnet_ids" "private" {
-  vpc_id = var.vpc_id
-
-  tags = {
-    tier = "private"
-  }
+locals {
+  managed_node_group_name = "deployment"
 }
 
 module "eks" {
-  source          = "terraform-aws-modules/eks/aws"
-  version         = "17.24.0"
+  source  = "terraform-aws-modules/eks/aws"
+  version = "~> 20.8"
+
   cluster_name    = var.cluster_name
-  cluster_version = "1.23"
-  subnets         = data.aws_subnet_ids.private.ids
-  enable_irsa     = true
-  vpc_id          = var.vpc_id
+  cluster_version = var.kubernetes_version
+
+  enable_irsa = true
+
+  enable_cluster_creator_admin_permissions = true
 
-  cluster_enabled_log_types = ["api", "scheduler", "controllerManager"] # https://docs.aws.amazon.com/eks/latest/userguide/control-plane-logs.html
+  access_entries = var.cluster_access_entries
 
-  node_groups_defaults = {
+  vpc_id     = var.vpc_id
+  subnet_ids = var.subnet_ids
+
+
+  # https://docs.aws.amazon.com/eks/latest/userguide/control-plane-logs.html
+  cluster_enabled_log_types = ["api", "scheduler", "controllerManager"]
+
+  cluster_endpoint_public_access = true
+
+  eks_managed_node_group_defaults = {
     ami_type  = "AL2_x86_64"
     disk_size = 50
   }
 
-  workers_group_defaults = {
-    protect_from_scale_in = true
```

Comment:
not sure why this was enabled.
[docs.aws.amazon.com/autoscaling/ec2/userguide/ec2-auto-scaling-instance-protection.html](https://docs.aws.amazon.com/autoscaling/ec2/userguide/ec2-auto-scaling-instance-protection.html)